### ISSUE TITLE
2.0 : static methods of the Query / Index class can be visible to all sub classes

### ISSUE THREAD

**Issue by pasin** [couchbase/couchbase-lite-ios#2051]
This is the same issue as #1987. The #1987 was resolved by having `ExpressionProtocol` as a based return type. However with Query / Index (especially Query), having something like `QueryProtocol` as a returned type doesn't feel intuitive giving the fact that developers tend to create an iVar for the Query object. After getting opinions from the team, we can solve this cross-platform and it makes more sense to separate the factory class from the returning class / interface.

To sum up, we could have something like this `let query: Query = QueryBuilder.select(....)`
